src/FilesPanel.jl

```diff
@@ -178,28 +237,42 @@ function get_selected_file(treeview::GtkTreeView,list::GtkTreeStore)
         return nothing
     end
 end
-function file_already_opened(filename)
+function file_tab_index(filename)
     for i=1:length(editor)
         t = get_tab(editor,i)
         if typeof(t) == EditorTab && t.filename == filename #in case we want to have something else in the editor
-            return true
+            return i
         end
     end
-    return false
+    return -1
 end
 function open_file(treeview::GtkTreeView,list::GtkTreeStore)
     file = get_selected_file(treeview,list)
-    if file != nothing && !file_already_opened(file)
+    file_idx = file_tab_index(file)
+    if file != nothing && file_idx==-1
         open_in_new_tab(file)
+    else
+        set_current_page_idx(editor,file_idx)
+    end
+end
+function nearest_folder_from_current_iterator(filespanel)
+    if filespanel.list[filespanel.current_iterator,5] == type_file
+        #Get parent folder
+        iterator = copy_up(GtkTreeModel(filespanel.list),filespanel.current_iterator)
+    else
+        iterator = filespanel.current_iterator
     end
+    return iterator
 end
 #=File path menu =#
-function path_dialog_create_file_cb(ptr::Ptr, data)
+@guarded nothing function path_dialog_create_file_cb(ptr::Ptr, data)
     (te_filename, filespanel) = data
     #TODO check overwrite
+
+    iterator = nearest_folder_from_current_iterator(filespanel)
     filename = getproperty(te_filename, :text, AbstractString)
     touch(filename)
-    add_file(filespanel.list, dirname(filename), basename(filename), filespanel.current_iterator)
+    update!(filespanel.list, filename, iterator)
     open_in_new_tab(filename)
     hide(filespanel.path_dialog.dialog)
```

Comment:
You could use visible(filespanel.path_dialog.dialog,false) here. It does the same thing (I guess that's why gtk_widget_hide wasn't implemented).